### ISSUE TITLE
Why key-value pair is not serialized as dictionary?

### ISSUE THREAD

**Issue by FixRM** [albanian-xrm/Xrm-Entity-Serializer#5]
Hi @BetimBeja,
As I see, this serializer gives similar result as default object serializer. I mean that key-value pair is not serialized as dictionary as it was expected by me. I guess my expectation was wrong, but what was the idea behind it, and is there any option to have result like the following?:

```json
{
    "id" : "***",
    "logicalName": "account",
    "name" : "account name"
}
```

**Comment by BetimBeja** [comment]
Hello @FixRM ,
the main idea behind this serializer is to be able to serialize the Entities to json and back. The main problem to solve is when the specified type is object, and I can only solve it by specifying the "$type" property. Maybe something can be done to guess the type of the object that you specified as EntityReference, but this behavior was not part of the current implementation.

**Comment by FixRM** [comment]
No, I'm not talking about EntityReference, this was a shrort example of Entity itself. In the times of OData v2 endpoint, we used to deal with similar jsons:
```json
{
    "entityid" : id,
    "field1name" : field1value,
    "field2name" : field2value,
    "field3name" : field3value,
    "field4name" : field4value,
}
```
It's kind of weird to deal with
```json
"Id":{
            "$type":"System.Guid, mscorlib",
            "$value":"f2944452-c67e-e811-90fe-005056b1c5f1"
        },
        "LogicalName":"team",
        "Attributes":{
            "$type":"Microsoft.Xrm.Sdk.AttributeCollection, Microsoft.Xrm.Sdk",
            "$value":[
                {
                    "Key":"name",
                    "Value":"A-TEAM"
                },
                {
                    "Key":"teamid",
                    "Value":{
                        "$type":"System.Guid, mscorlib",
                        "$value":"f2944452-c67e-e811-90fe-005056b1c5f1"
                    }
                },
```
with js... It's not clear how to take property value from there.

At the time of writing I guess that you build this serializer not with ASP/WEB stuff in mind :)

**Comment by BetimBeja** [comment]
You can always write a conversion function in javascript, so that you can transform the "type safe" JSON to the format that is easier to use for you. I have functions which map to class-es so that I preserve the "type" information and can correctly roundtrip this information between javascript and .net.
